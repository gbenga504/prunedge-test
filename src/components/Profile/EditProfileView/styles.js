import { commonCss, fontsize } from "../../../Css";

export const styles = {
  container: {
    ...commonCss.flexColumn,
    padding: 22,
    background: "#fff",
    width: "52.5%",
    borderRadius: 4,
    marginLeft: "1.5%",
    height: "max-content",
    boxShadow: "0px 2px 24px rgba(0, 0, 0, 0.04)",
    marginBottom: 20,
    paddingBottom: 40
  },
  title: {
    ...commonCss.title
  },
  marker: {
    ...commonCss.titleMarker
  },
  row: {
    ...commonCss.flex
  },
  submitButton: {
    display: "flex",
    alignSelf: "flex-end",
    marginTop: 36,
    width: 100,
    textTransform: "capitalize"
  },
  menuItemRoot: {
    fontSize: fontsize.base
  }
};
