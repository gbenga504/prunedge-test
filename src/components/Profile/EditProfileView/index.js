import React from "react";
import withStyles from "@material-ui/core/styles/withStyles";
import Typography from "@material-ui/core/Typography";
import MenuItem from "@material-ui/core/MenuItem";
import Button from "@material-ui/core/Button";

import { styles } from "./styles";
import Input from "../../../atoms/Input";

class EditProfileView extends React.PureComponent {
  state = {
    fullName: "",
    email: "",
    phoneNumber: "",
    gender: "male",
    position: "ui/ux designer",
    homeAddress: ""
  };

  handleChange = (name, value) => {
    this.setState({ [name]: value });
  };

  render() {
    let { classes } = this.props,
      inputTopSpacing = { marginTop: 12 };

    return (
      <div className={classes.container}>
        <div style={{ position: "relative", width: "auto", marginBottom: 40 }}>
          <Typography variant="body1" classes={{ body1: classes.title }}>
            Edit Profile
          </Typography>
          <span className={classes.marker} />
        </div>
        <Input
          label="Full name"
          type="text"
          name="fullName"
          required={true}
          onChange={this.handleChange}
        />
        <Input
          label="Email"
          containerStyle={inputTopSpacing}
          type="email"
          name="email"
          required={true}
          onChange={this.handleChange}
        />
        <Input
          label="Phone Number"
          containerStyle={inputTopSpacing}
          type="tel"
          name="phoneNumber"
          required={true}
          onChange={this.handleChange}
        />
        <div className={classes.row}>
          <Input
            label="Gender"
            halfInput={true}
            containerStyle={{ ...inputTopSpacing, marginRight: 5 }}
            type="text"
            select={true}
            value={this.state.gender}
            name="gender"
            required={true}
            onChange={this.handleChange}
          >
            <MenuItem value="male" classes={{ root: classes.menuItemRoot }}>
              Male
            </MenuItem>
            <MenuItem value="female" classes={{ root: classes.menuItemRoot }}>
              Female
            </MenuItem>
          </Input>
          <Input
            label="Position"
            halfInput={true}
            containerStyle={{ ...inputTopSpacing, marginLeft: 5 }}
            type="text"
            select={true}
            name="position"
            value={this.state.position}
            required={true}
            onChange={this.handleChange}
          >
            <MenuItem
              value="ui/ux designer"
              classes={{ root: classes.menuItemRoot }}
            >
              UI/UX Designer
            </MenuItem>
            <MenuItem
              value="frontend software engineer"
              classes={{ root: classes.menuItemRoot }}
            >
              FrontEnd Software Engineer
            </MenuItem>
            <MenuItem
              value="backend software engineer"
              classes={{ root: classes.menuItemRoot }}
            >
              BackEnd Software Engineer
            </MenuItem>
          </Input>
        </div>
        <Input
          multiline={true}
          label="Home Address"
          rows="5"
          rowsMax="3"
          maxLength="500"
          containerStyle={inputTopSpacing}
          type="text"
          name="homeAddress"
          required={true}
          onChange={this.handleChange}
        />
        <Button
          variant="contained"
          color="primary"
          className={classes.submitButton}
        >
          Update
        </Button>
      </div>
    );
  }
}

export default withStyles(styles)(EditProfileView);
