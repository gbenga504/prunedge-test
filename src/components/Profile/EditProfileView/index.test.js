import React from "react";
import { create } from "react-test-renderer";

import EditProfileView from "./index";

const createNodeMock = el => {
  if (el.type === "textarea") {
    const mockEl = document.createElement("textarea");
    mockEl.style.padding = "0";
    return mockEl;
  }
  return null;
};

describe("EditProfileView", () => {
  it("renders the edit profile view", () => {
    const tree = create(<EditProfileView />, { createNodeMock });
    expect(tree).toMatchSnapshot();
  });
});
