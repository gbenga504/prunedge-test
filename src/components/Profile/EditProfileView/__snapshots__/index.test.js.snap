// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfileView renders the edit profile view 1`] = `
<div
  className="EditProfileView-container-1"
>
  <div
    style={
      Object {
        "marginBottom": 40,
        "position": "relative",
        "width": "auto",
      }
    }
  >
    <p
      className="MuiTypography-root MuiTypography-body1 EditProfileView-title-2"
    >
      Edit Profile
    </p>
    <span
      className="EditProfileView-marker-3"
    />
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "width": "100%",
      }
    }
  >
    <p
      className="MuiTypography-root MuiTypography-body2 Input-label-37"
    >
      Full name
    </p>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        onClick={[Function]}
        style={
          Object {
            "fontSize": 14,
            "height": 45,
          }
        }
      >
        <input
          aria-invalid={false}
          aria-label="input"
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          rows=""
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-76 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-77"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "marginTop": 12,
        "width": "100%",
      }
    }
  >
    <p
      className="MuiTypography-root MuiTypography-body2 Input-label-37"
    >
      Email
    </p>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        onClick={[Function]}
        style={
          Object {
            "fontSize": 14,
            "height": 45,
          }
        }
      >
        <input
          aria-invalid={false}
          aria-label="input"
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          rows=""
          type="email"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-76 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-77"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "marginTop": 12,
        "width": "100%",
      }
    }
  >
    <p
      className="MuiTypography-root MuiTypography-body2 Input-label-37"
    >
      Phone Number
    </p>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        onClick={[Function]}
        style={
          Object {
            "fontSize": 14,
            "height": 45,
          }
        }
      >
        <input
          aria-invalid={false}
          aria-label="input"
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          rows=""
          type="tel"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-76 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-77"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    className="EditProfileView-row-4"
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "marginRight": 5,
          "marginTop": 12,
          "width": "50%",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body2 Input-label-37"
      >
        Gender
      </p>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
          style={
            Object {
              "fontSize": 14,
              "height": 45,
            }
          }
        >
          <div
            aria-haspopup="listbox"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense MuiSelect-outlined"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Male
          </div>
          <input
            aria-label="input"
            rows=""
            type="hidden"
            value="male"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-76 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-77"
              style={
                Object {
                  "width": 0,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "marginLeft": 5,
          "marginTop": 12,
          "width": "50%",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body2 Input-label-37"
      >
        Position
      </p>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
          style={
            Object {
              "fontSize": 14,
              "height": 45,
            }
          }
        >
          <div
            aria-haspopup="listbox"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense MuiSelect-outlined"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            UI/UX Designer
          </div>
          <input
            aria-label="input"
            rows=""
            type="hidden"
            value="ui/ux designer"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-76 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-77"
              style={
                Object {
                  "width": 0,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "marginTop": 12,
        "width": "100%",
      }
    }
  >
    <p
      className="MuiTypography-root MuiTypography-body2 Input-label-37"
    >
      Home Address
    </p>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        onClick={[Function]}
        style={
          Object {
            "fontSize": 14,
            "height": false,
          }
        }
      >
        <textarea
          aria-invalid={false}
          aria-label="input"
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          rows="5"
          style={
            Object {
              "height": undefined,
              "overflow": null,
            }
          }
        />
        <textarea
          aria-hidden={true}
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          readOnly={true}
          style={
            Object {
              "height": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "visibility": "hidden",
            }
          }
          tabIndex={-1}
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-76 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-77"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root EditProfileView-submitButton-5 MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Update
    </span>
  </button>
</div>
`;
